addons/stock/models/stock_quant.py

```diff
@@ -498,8 +622,10 @@ def _unlink_zero_quants(self):
         """
         precision_digits = max(6, self.sudo().env.ref('product.decimal_product_uom').digits * 2)
         # Use a select instead of ORM search for UoM robustness.
-        query = """SELECT id FROM stock_quant WHERE (round(quantity::numeric, %s) = 0 OR quantity IS NULL) AND round(reserved_quantity::numeric, %s) = 0;"""
-        params = (precision_digits, precision_digits)
+        query = """SELECT id FROM stock_quant WHERE (round(quantity::numeric, %s) = 0 OR quantity IS NULL)
+                                                     AND round(reserved_quantity::numeric, %s) = 0
```

Comment:
`AND (round(reserved_quantity::numeric, %s) = 0 OR reserved_quantity IS NULL)` to prevent any issue